eZ/Publish/API/Repository/Tests/SearchServiceFulltextEmbedTest.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\API\Repository\Tests;
+
+use eZ\Publish\API\Repository\SearchService;
+use eZ\Publish\API\Repository\Values\Content\LocationQuery;
+use eZ\Publish\API\Repository\Values\Content\Query;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+use eZ\Publish\API\Repository\Values\Content\Search\SearchHit;
+use eZ\Publish\Core\FieldType\RichText\Value as RichTextValue;
+use eZ\Publish\Core\Repository\Values\Content\Content;
+use eZ\Publish\Core\Repository\Values\Content\ContentCreateStruct;
+
+/**
+ * Test case for full text search in the SearchService (for embed).
+ *
+ * @see \eZ\Publish\API\Repository\SearchService
+ * @group integration
+ * @group search
+ * @group fulltext
+ */
+class SearchServiceFulltextEmbedTest extends BaseTest
+{
+    private const SLAVE_ARTICLE_NAME = 'test1';
+
+    private static $createdIds = [];
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $repository = $this->getRepository(false);
+
+        if (false === $repository->getSearchService()->supports(SearchService::CAPABILITY_ADVANCED_FULLTEXT)) {
+            $this->markTestSkipped('Engine says it does not support advance fulltext format');
+        }
+    }
+
+    public function testPrepareContent(): void
+    {
+        $contentService = $this->getRepository()->getContentService();
+        $baseArticleStruct = $this->prepareBaseArticleStruct();
+
+        $slaveArticleStruct = $this->fillSlaveArticleStruct(clone $baseArticleStruct);
+        $slaveArticleContent = $contentService->publishVersion(
+            $this->createContent($slaveArticleStruct)->versionInfo
+        );
+
+        $mainArticleStruct = $this->fillMainArticleStruct(clone $baseArticleStruct, $slaveArticleContent->id);
+        $mainArticleContent = $contentService->publishVersion(
+            $this->createContent($mainArticleStruct)->versionInfo
+        );
+
+        $this->refreshSearch($this->getRepository());
+
+        self::$createdIds = [
```

Comment:
This is not correct per PHPUnit rules. A test is a method which asserts something (you should actually see Risky warning here). If you need to prepare some data, make a method which does not start with `test...`. Usually those are data providers, but because the code here deals with persistence, I wouldn't advise that. Simple method.

Reply:
You're right, that isn't the best solution, but there are some reasons to done it this way:
I need to run this once, because of performance reasons and in case of that has an impact on the search result. If you run this twice the search result will return 4 hits instead of 2. Maybe this is not so much problem, because of "gratherThanOrEqual: 2" assertion, but this type of assertion was used by another reasons. I cannot also use the "setUpBeforeClass()" template method, cause of the static nature of it, while I need an object context (this). There's also at least one way more: I can add the "hasPreparedContent" state and check it before every test and run prepare method if needed. What do you think about this solution @alongosz ?